neuralprophet/plot_utils.py

```diff
@@ -542,3 +557,84 @@ def get_valid_configuration(  # move to utils
             "components_list": plot_components,
         }
     return valid_configuration
+
+
+def validate_current_env_for_resampler(auto: bool = False) -> Optional[bool]:
+    """
+    Validate the current environment to check if it is a valid environment for plotly-resampler and if invalid trigger warning message.
+
+    Parameters
+    ----------
+    auto: bool, optional
+        If True, the function will automatically switch to a valid environment if the current environment is not valid.
+        If False, the function will return None if the current environment is not valid.
+    Returns
+    -------
+    bool :
+        True if the current environment is a valid environment to run the code, False if the current environment is
+        not a valid environment to run the code. None if the current environment is not a valid environment to run
+        the code and the function did not switch to a valid environment.
+    """
+
+    from IPython import get_ipython
+
+    if "google.colab" in str(get_ipython()):
+        if auto:
+            log_warning_resampler_switch_to_valid_env()
+            valid_env = False
+        else:
+            log_warning_resampler_invalid_env()
+            valid_env = None
+    else:
+        if is_notebook():
+            valid_env = True
+        else:
+            if auto:
+                log_warning_resampler_switch_to_valid_env()
+                valid_env = False
+            else:
+                log_warning_resampler_invalid_env()
+                valid_env = None
+    return valid_env
+
+
+def is_notebook():
+    """
+    Determine if the code is being executed in a Jupyter notebook environment.
+
+    Returns
+    -------
+    bool :
+        True if the code is being executed in a Jupyter notebook, False otherwise.
+    """
+    try:
+        from IPython import get_ipython
+
+        if "IPKernelApp" not in get_ipython().config:  # pragma: no cover
+            return False
+    except ImportError:
+        return False
+    except AttributeError:
+        return False
+    return True
+
+
+def auto_set_plotting_backend(plotting_backend):
+    """Automatically set the plotting backend if it is not specified and triggers warning messages if the current
+    environment is not valid. If the plotting backend is specified as "plotly-resampler", triggers warning message.
+
+    Parameters
+    ----------
+    plotting_backend: str
+        The plotting backend to use.
+
+    Returns
+    -------
+    str
+        The new plotting backend.
+    """
+    if plotting_backend is None:
+        plotting_backend = "plotly-resampler" if validate_current_env_for_resampler(auto=True) else "plotly"
+    if plotting_backend == "plotly-resampler":
```

Comment:
```suggestion
    elif plotting_backend == "plotly-resampler":
```
Does not make sense to run `validate_current_env_for_resampler` again if we just used it with `auto=True` or?